### ISSUE TITLE
source code?

### ISSUE THREAD

**Issue by tony--** [almenjonatan/terraforming-mars-gamelog#1]
I came here from your [Steam community post](https://steamcommunity.com/app/800270/discussions/0/3103515720272474639/?tscn=1615437133#c3094509155619435452).  I'm interested, but reluctant to download dlls without source code.  I'm sure many others must feel the same way.  If you share the project so that we could build the dlls ourselves it would help.

Also, posting the link to the Melon Loader installer without any explanation is another red flag for those not familiar with Melon Loader.  I suggest you add a sentence explaining that you use Melon Loader and what it is, maybe with a link to the wiki and/or the subreddit.

I get that you just want to share this and the extra effort might not be worth it to you, just sharing some feedback as an interested potential user :)

If you'd like, I'd be happy to open a pull request to propose README edits after you share the source code and I can build it for myself.

Thanks for reading!

**Comment by adamk33n3r** [comment]
I concur here. I'm a software developer so I'm not going to just trust dll's blindly. Would love to have the source code available for this. Would love to have this mod since the base game has no way of knowing what happened after its over.

**Comment by almenjonatan** [comment]
Hi, sorry for the late response.

We are probably going to to open source it, we have opted to keep it as closed source since this mod makes it possible to cheat at the game.

If you would like take a look before its released you can run through a disassembler.

**Comment by adamk33n3r** [comment]
The cheating scenario makes since, wonder if you could make an option to only output at the end of the game?

**Comment by almenjonatan** [comment]
Yeah it's only writing to the gamelog after  the final plant conversion.

The problem is that player and game state is stored at client side, all information is available to everyone.

**Comment by tony--** [comment]
So is the cheating you are concerned about peeking at opponent's hand?  Can't think of anything else you could do with the game state.  AFAIK your hand is the only hidden information.

**Comment by almenjonatan** [comment]
You could also manipulate the game, such as terraforming the whole planet in one generation, infinite money and probably a lot more! :)

**Comment by tony--** [comment]
Ah, so it syncs your local game state to the other clients after you take a turn?  Yuck.
